graylog-storage-opensearch2/src/main/java/org/graylog/storage/opensearch2/RemoteReindexingMigrationAdapterOS2.java

```diff
@@ -266,28 +238,35 @@ private String getErrors(GetTaskResponse task) {
     }
 
     @Override
-    public IndexerConnectionCheckResult checkConnection(URI uri, String username, String password, boolean trustUnknownCerts) {
+    public IndexerConnectionCheckResult checkConnection(@Nonnull URI remoteHost, @Nullable String username, @Nullable String password, @Nullable String allowlist, boolean trustUnknownCerts) {
+
+
         try {
-            final List<String> discoveredIndices = getAllIndicesFrom(uri, username, password, trustUnknownCerts);
+            final RemoteReindexAllowlist reindexAllowlist = new RemoteReindexAllowlist(remoteHost, allowlist);
+            reindexAllowlist.validate();
+            final List<String> discoveredIndices = getAllIndicesFrom(remoteHost, username, password, trustUnknownCerts);
             return IndexerConnectionCheckResult.success(discoveredIndices);
         } catch (Exception e) {
             return IndexerConnectionCheckResult.failure(e);
         }
     }
 
-    private boolean isRemoteReindexAllowed(List<String> allowlistEntries) {
-        final String allowlistSettingValue = client.execute((restHighLevelClient, requestOptions) -> {
+    private boolean isRemoteReindexAllowed(List<String> allowlistEntries, String clusterAllowlistSetting) {
```

Comment:
method can be removed, provided by `RemoteReindexAllowList`

Reply:
Thx, removed!